File: pushk_engagelab_orig/src/main/java/com/engagelab/app/activity/MainActivity.java
package com.engagelab.app.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.engagelab.privates.common.log.MTCommonLog;

import com.engagelab.app.R;
import com.engagelab.app.listener.OnStatusListener;
import com.engagelab.app.listener.StatusObserver;
import com.engagelab.app.utils.ExampleUtil;
import com.engagelab.privates.core.api.MTCorePrivatesApi;

/**
 * 主要用于初始化
 */
public class MainActivity extends Activity implements View.OnClickListener, OnStatusListener {

    private static final String TAG = "MainActivity";
    private TextView tvUid;
    private TextView tvRid;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    @Override
    protected void onResume() {
        super.onResume();
        tvUid.setText(MTCorePrivatesApi.getUserId(this.getApplicationContext()) + "");
        tvRid.setText(MTCorePrivatesApi.getRegistrationId(this.getApplicationContext()));
        // 监听状态变化
        StatusObserver.getInstance().addListener(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        // 监听移除
        StatusObserver.getInstance().removeListener();
    }

    private void initView() {
        findViewById(R.id.btn_push).setOnClickListener(this);

        TextView tvPackageName = findViewById(R.id.tv_package_name);
        TextView tvAppKey = findViewById(R.id.tv_app_key);
        TextView tvAppChannel = findViewById(R.id.tv_app_channel);
        TextView tvAppProcess = findViewById(R.id.tv_app_process);
        TextView tvUserService = findViewById(R.id.tv_user_service);
        TextView tvUserReceiver = findViewById(R.id.tv_user_receiver);
        tvUid = findViewById(R.id.tv_uid);
        tvRid = findViewById(R.id.tv_rid);

        tvPackageName.setText(this.getApplicationContext().getPackageName());
        tvAppKey.setText(ExampleUtil.getAppKey(this.getApplicationContext()));
        tvAppChannel.setText(ExampleUtil.getAppChannel(this.getApplicationContext()));
        tvAppProcess.setText(ExampleUtil.getAppProcess(this.getApplicationContext()));
        tvUserService.setText(ExampleUtil.getCommonServiceName(this.getApplicationContext()));
        tvUserReceiver.setText(ExampleUtil.getCommonReceiverName(this.getApplicationContext()));
    }

    @Override
    public void onNotificationStatus(boolean status) {
        // do nothing
    }

    @Override
    public void onConnectStatus(boolean status) {
        if (status) {
            tvUid.setText(MTCorePrivatesApi.getUserId(this.getApplicationContext()) + "");
            tvRid.setText(MTCorePrivatesApi.getRegistrationId(this.getApplicationContext()));
        }
    }

    @Override
    public void onClick(View view) {
        if (view.getId()==R.id.btn_push){
            if (TextUtils.isEmpty(ExampleUtil.getCommonServiceName(this.getApplicationContext()))) {
                MTCommonLog.e(TAG, "userService is null, please create new Service extends MTCommonService");
                Toast.makeText(this.getApplicationContext(), "userService is null, please create new Service extends MTCommonService", Toast.LENGTH_SHORT).show();
                return;
            }
            if (TextUtils.isEmpty(ExampleUtil.getCommonReceiverName(this.getApplicationContext()))) {
                MTCommonLog.e(TAG, "userReceiver is null, please create new BroadcastReceiver extends MTCommonReceiver");
                Toast.makeText(this.getApplicationContext(), "userReceiver is null, please create new BroadcastReceiver extends MTCommonReceiver", Toast.LENGTH_SHORT).show();
                return;
            }
            Intent pushIntent = new Intent(this, PushActivity.class);
            startActivity(pushIntent);
        }
    }
}

File: pushk_engagelab_test/src/main/java/com/mozhimen/pushk/engagelab/test/activity/MainActivity.kt
package com.mozhimen.pushk.engagelab.test.activity

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import com.engagelab.privates.common.log.MTCommonLog
import com.engagelab.privates.core.api.MTCorePrivatesApi
import com.engagelab.privates.push.api.CustomMessage
import com.mozhimen.bindk.bases.viewbinding.activity.BaseActivityVBVM
import com.mozhimen.bindk.bases.viewdatabinding.activity.BaseActivityVDBVM
import com.mozhimen.kotlin.elemk.android.content.cons.CIntent
import com.mozhimen.kotlin.utilk.android.content.startContext
import com.mozhimen.kotlin.utilk.android.widget.showToast
import com.mozhimen.pushk.engagelab.PushKEngagelab
import com.mozhimen.pushk.engagelab.commons.IMTCommonListener
import com.mozhimen.pushk.engagelab.test.MainViewModel
import com.mozhimen.pushk.engagelab.test.component.CustomMessageActivity34X
import com.mozhimen.pushk.engagelab.test.databinding.ActivityMainBinding
import com.mozhimen.pushk.engagelab.test.utils.ExampleUtil

/**
 * @ClassName MainActivity
 * @Description TODO
 * @Author mozhimen
 * @Date 2024/12/3
 * @Version 1.0
 */
/**
 * 主要用于初始化
 */
class MainActivity : BaseActivityVDBVM<ActivityMainBinding, MainViewModel>(), IMTCommonListener {
    @SuppressLint("SetTextI18n")
    override fun onResume() {
        super.onResume()
        vm.liveUid.value = MTCorePrivatesApi.getUserId(this.applicationContext).toString()
        vm.liveRid.value = MTCorePrivatesApi.getRegistrationId(this.applicationContext)
        // 监听状态变化
        PushKEngagelab.instance.addMTCommonListener(this)
    }

    override fun onPause() {
        // 监听移除
        PushKEngagelab.instance.removeMTCommonListener(this)
        super.onPause()
    }

    override fun initView(savedInstanceState: Bundle?) {
        vdb.btnPush.setOnClickListener {
            if (TextUtils.isEmpty(ExampleUtil.getCommonServiceName())) {
                MTCommonLog.e(TAG, "userService is null, please create new Service extends MTCommonService")
                "userService is null, please create new Service extends MTCommonService".showToast()
                return@setOnClickListener
            }
            if (TextUtils.isEmpty(ExampleUtil.getCommonReceiverName())) {
                MTCommonLog.e(TAG, "userReceiver is null, please create new BroadcastReceiver extends MTCommonReceiver")
                "userReceiver is null, please create new BroadcastReceiver extends MTCommonReceiver".showToast()
                return@setOnClickListener
            }
            startContext<PushActivity>()
        }
    }

    override fun bindViewVM(vdb: ActivityMainBinding) {
        vdb.vm = vm
    }

    override fun onCustomMessage(context: Context, customMessage: CustomMessage) {
        // 用于演示自定义消息展示
        startContext<CustomMessageActivity34X> {
            setFlags(CIntent.FLAG_ACTIVITY_NEW_TASK)
            putExtra("message", customMessage)
        }
    }

    @SuppressLint("SetTextI18n")
    override fun onConnectStatus(context: Context, enable: Boolean, registrationId: String?) {
        if (enable) {
            vm.liveUid.value = MTCorePrivatesApi.getUserId(this.applicationContext).toString()
            vm.liveRid.value = MTCorePrivatesApi.getRegistrationId(this.applicationContext)
        }
    }
}